packages/flutter_tools/lib/src/asset.dart

```diff
@@ -113,8 +113,9 @@ class AssetBundle {
         manifestDescriptor['uses-material-design'];
 
     for (_Asset asset in assetVariants.keys) {
-      assert(asset.assetFileExists);
-      entries[asset.assetEntry] = new DevFSFileContent(asset.assetFile);
+      assert(asset.assetFileExists || assetVariants.isNotEmpty);
```

Comment:
The original version of this check had been an invariant but it probably should throw an error. With a mighty helpful error message :-).